lib/resources/hive_storage.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:hive_flutter/hive_flutter.dart';
+import 'package:qubic_wallet/helpers/app_logger.dart';
+import 'package:qubic_wallet/models/transaction_vm.dart';
+
+enum HiveBoxesNames {
+  storedTransactions,
+}
+
+class HiveStorage {
+  HiveStorage() {
+    _init();
+  }
+  late final Box<TransactionVm> storedTransactions;
+
+  Future<void> _init() async {
+    await Hive.initFlutter();
+    Hive.registerAdapter(TransactionVmAdapter());
+    openTransactionsBox();
+  }
+
+  Future<void> openTransactionsBox() async {
+    storedTransactions = await Hive.openBox<TransactionVm>(
+        HiveBoxesNames.storedTransactions.name);
+  }
+
+  void addStoredTransaction(TransactionVm transactionVm) {
+    storedTransactions.put(transactionVm.id, transactionVm);
+  }
+
+  void removeStoredTransaction(String transactionId) {
```

Comment:
@ahmed-tarek-salem  I see this is only invoked when the user taps "Delete".
Shouldn't also be invoked for the local trx that were already returned by the network call?

Reply:
Resolved.